src/components/element-editor/basic-details/common/type-field.tsx

```diff
@@ -27,8 +27,18 @@ export const TypeField = () => {
         value: EElementType,
         field: ControllerRenderProps<TFormFieldValues, "type">
     ) => {
+        if (value === field.value) {
+            return;
+        }
         selectElementType(value);
         setValue("validation", []);
+        /** The purpose of this setValue is to deliberately clear the preselectedValue
+         * when the input type of the preselectedValue is different. Currently,
+         * the preselectedValue is different only when the input type is dropdown.
+         */
+        if (field.value === EElementType.DROPDOWN) {
+            setValue("preselectedValue", "");
+        }
```

Comment:
is it possible to move this logic into `DropdownBasicDetails`?
because this logic has cross cutting concerns and it may not be obvious enough to be kept here

Reply:
This logic currently runs when `type` is being toggled from `EElementType.DROPDOWN` to another element. I did try running it on unmounting of `DropdownPreselectedValue` with a `useEffect`, but faced the same issue of the value being retained in the input field.

Reply:
yes a useEffect in `BasicDetails` would work. Will shift it there. As for the 2nd part, can I clarify what you mean? Currently based on Karling's mockup, we will reset the fields even if the form is clean and in saved state.

Reply:
because I am proposing to create a useEffect with only type as dependency:
```
 useEffect(() => {
  // reset
}, [type]);
```

we cant reset the form based on type alone because it will also run every time I select or switch an element in the main panel. but since the form will be dirtied on changing the type, we can reset the form based on that.

Reply:
I see. I missed that out. Thank you for the clarification!